### PR TITLE
refactor: [M3-6672] - Refactor `<AddNewMenu />` to use MUI rather than Reach UI

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

Refactors `<AddNewMenu />` to use Material UI and simplifies styles and code in general

## Major Changes 🔄

- Makes this Menu match the style and behavior of our other MUI menus
- Takes steps to get `@reach/*` out of the repo 🎉🗑️
  - This allows us to better rely on our theme instead of having to one-off style all of the Reach UI components
  - As we get rid of packages like `react-select` and `@reach/*`, our components will naturally become more consistant because they will pull from our MUI theme

## Preview 📷

| Before  | After   |
| ------- | ------- |
| Screenshot 2023-06-07 at 1 38 53 PM | Screenshot 2023-06-07 at 1 38 48 PM |

## How to test 🧪
- Test the UI and the functionality of the `<AddNewMenu />`

### CODE REVIEW

**Comment by bnussman** [comment]
Your PR has a changeset ✅

**Review by jaalah-akamai — APPROVED** [review]
Tested, the new menu looks good

File: packages/manager/src/features/TopMenu/AddNewMenu/AddNewMenu.tsx [hunk 1]
```diff
@@ -10,243 +10,156 @@ import NodebalancerIcon from 'src/assets/icons/entityIcons/nodebalancer.svg';
...
+    setAnchorEl(null);
+  };
 
-const useStyles = makeStyles((theme: Theme) => ({
-  wrapper: {
-    position: 'relative',
-    marginRight: theme.spacing(1),
-    [theme.breakpoints.down('md')]: {
-      flex: 1,
+  const links = [
```
Comment (hkhalil-akamai) [inline packages/manager/src/features/TopMenu/AddNewMenu/AddNewMenu.tsx:39]: I love this, super DRY

**Review by hkhalil-akamai — APPROVED** [review]
Tested, looks great and the animation is slick! Code cleanup is a plus. Might want a final approval from UX since there are some minor changes to the appearance of the menu.

**Comment by bnussman-akamai** [comment]
I'll double check with UX, thanks @hkhalil-akamai!

**Pull request merged**